nft-minter/tests/rust_test.rs

```diff
@@ -393,6 +395,57 @@ fn buy_whitelist_test() {
         .assert_ok();
 }
 
+#[test]
+fn repair_nft_test() {
+    let mut nm_setup = NftMinterSetup::new(nft_minter::contract_obj);
+    let first_tier = FIRST_TIERS[0];
+    let first_user_addr = nm_setup.first_user_address.clone();
+
+    nm_setup.create_default_brands();
+
+    nm_setup.b_mock.set_nft_balance_all_properties(
+        &first_user_addr,
+        FIRST_TOKEN_ID,
+        1,
+        &rust_biguint!(1),
+        &FIRST_ATTRIBUTES.to_vec(),
+        7_700,
+        None,
+        None,
+        None,
+        &uris_to_vec(FIRST_URIS),
+    );
+    nm_setup.b_mock.set_nft_balance_all_properties(
+        &first_user_addr,
+        FIRST_TOKEN_ID,
+        2,
+        &rust_biguint!(1),
+        &FIRST_ATTRIBUTES.to_vec(),
+        900,
+        None,
+        None,
+        None,
+        &uris_to_vec(FIRST_URIS),
+    );
+    nm_setup
+        .call_nft_upgrade(
```

Comment:
You should check the modified values.

Reply:
the only element changing within this endpoint is the royalties attribute and it is changed by a constant value